### ISSUE TITLE
Cluster Provisioning question(~azure)

### ISSUE THREAD

**Issue by helloravi** [deis/deis#4563]
My friend's startup has a Microsoft bizspark account. Microsoft offers 750 dollars of credits to five different developer accounts with each having 150USD worth of credits. They credits are **not aggregated** into one account.

Can we create an instance of a server in each of the accounts separately in the same region and provision the cluster? If it is possible, how would the installation process change?

A lot of start ups would be benefitted if this is possible.

I am a rails developer who mostly worked on Heroku.

**Comment by bacongobbler** [comment]
I think the question about provisioning the servers in the same region is a question best suited for Azure, but assuming that it's all generally the same, it should be possible to modify https://github.com/deis/deis/blob/c3381a0cca1add9fd62535e74f8f885911d75c51/contrib/azure/parameters.json#L24 to deploy 1 node per account, re-using the same user data for each account deployment.

**Comment by helloravi** [comment]
Thank you for taking the time to answer my question.

1. Extending this argument, could we use three machines one from digital ocean, one from AWS and the last one from Azure to create a cluster?  If not, what is preventing from this being possible(discovery url?). I am only asking this to make sure I understand what you are saying.

2. When we create a cluster, can all the machines be of different sizes as long as they meet the minimum system requirements?

**Comment by carmstrong** [comment]
The main issue here is that each use of the provision script will create a *new* Deis cluster. The Azure template will create a new private network for each cluster. Even if you use the script once per account, the hosts will be unable to communicate with each other.

The scenario you're describing is unsupported by using our provision scripts. 

> Extending this argument, could we use three machines one from digital ocean, one from AWS and the last one from Azure to create a cluster? If not, what is preventing from this being possible(discovery url?). I am only asking this to make sure I understand what you are saying.

Yes, the main requirement being that all of the hosts can communicate over a private network. This is something that you'd have to configure manually. Additionally, latency between hosts is an issue. If they are on different networks in different locations, you're likely going to run into issues.

In short, this topology probably isn't worth it. If you're trying to save money, perhaps deploying to Digital Ocean is a better bet.

> When we create a cluster, can all the machines be of different sizes as long as they meet the minimum system requirements?

Yes.

**Comment by helloravi** [comment]
@carmstrong - Thank you for very much for your clarifications. I have a few more questions.

1. Given that I can create the three nodes in the same location in Azure, would I be able to use the method suggested by @bacongobbler(modifying parameters.json and copying userdata) and provision a cluster even if it involves some manual work? (This would mean my friend could have 5 nodes with 7GB ram each on Azure ). Since they are in the same location, that would take care of the latency issue(?).

2. In the installation process for Bare metal servers it is mentioned I quote "the hosts running Deis must be able to communicate via a private network in one of the RFC 1918 or RFC 6598 private address spaces: 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16, or 100.64.0.0/10". Will this be a problem if I create the nodes as I mentioned in step 1?

**Comment by bacongobbler** [comment]
This is not something we'll be able to get to for the LTS release, and provisioning Deis v2 is done through installing a vanilla kubernetes cluster so I'm going to go ahead and close this.
